smartsim/_core/utils/telemetry/collector.py

```diff
@@ -238,16 +121,15 @@ class DBCollector(Collector):
     """A base class for collectors that retrieve statistics from an orchestrator"""
 
     def __init__(self, entity: JobEntity, sink: Sink) -> None:
-        """Initialize the collector"""
         super().__init__(entity, sink)
         self._client: t.Optional[redisa.Redis[bytes]] = None
-        self._address = _Address(
+        self._address = _DBAddress(
             self._entity.config.get("host", ""),
             int(self._entity.config.get("port", 0)),
         )
 
     async def _configure_client(self) -> None:
-        """Configure and connect to the target database"""
+        """Configure the client connectoin to the target database"""
         try:
```

Comment:
Because my code is not very sophisticated, I start running telemetry collection immediately upon creation of the Collectors. There is a period of time at the beginning where an Orchestrator is starting up that collection will fail to connect. 

I know it's a weakness but don't want to add delayed start or triggering events so I traded off that complexity for letting these exceptions occur for a while.

The bad part of the trade off is that some collector MIGHT now fail forever. I previously built a whole callback mechanism where collectors would shut themselves off if they failed more than X times in a row, but removed it.  It wasn't in the core problem and I wasn't yet sure if it met my needs....

Great eye noticing this behavior.